### PR TITLE
add emacs-lsp integration example

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,6 +127,21 @@ Add the following elisp code to your configuration. (using `use-package`)
   (nix-mode . eglot-ensure))
 ```
 
+### Emacs with [`lsp-mode`]
+
+[`lsp-mode`]: https://github.com/emacs-lsp/lsp-mode
+
+``` jsonc
```
Comment (oxalica) [inline README.md:134]: This is definitely not `jsonc`.

**Review by oxalica — CHANGES_REQUESTED** [review]
Could you also add an example for LSP configurations?

**Comment by Mic92** [comment]
What do you mean by `LSP configurations`? The configuration was all it needed to get it running for me.

**Comment by oxalica** [comment]
> What do you mean by `examples for LSP configurations`? The configuration was all it needed to get it running for me.

Not just "get it running". There are some configuration options of the server, like formatting command. Usually LSP clients have a way to pass these configurations to the server.

To reduce questions like "why formatting is not working", I included the way to set these configurations for editors/plugins that I'm familiar with, like `coc.nvim` in the README https://github.com/oxalica/nil#vimneovim-with-cocnvim

**Comment by Mic92** [comment]
Maybe someone else can take this one over. I don't want to dig too deep into fiddling with emacs-lsp configuration and just wait for the next release to end up in doom-emacs so it just works out-of-the-box.

**Pull request closed without merging**